app/pktgen-main.c

```diff
@@ -488,6 +490,7 @@ main(int argc, char **argv)
     if (get_lcore_rxcnt(pktgen.l2p, i) || get_lcore_txcnt(pktgen.l2p, i)) {
         cli_printf("*** Error can not use initial lcore for a port\n");
         cli_printf("    The initial lcore is %d\n", rte_get_main_lcore());
+        scrn_destroy();
```

Comment:
Could add a goto label at line 555 and have this one goto label. At line 555 I try to cleanup everything, but creating a function to do the cleanup could work as well.